frontend/app/assets/javascripts/spree/frontend/checkout/address.js

```diff
@@ -15,8 +15,9 @@ Spree.ready(function ($) {
                 var x_states = []
                 for(var i = 0; i < json.length; i++) {
                     var obj = json[i];
-                    x_states.push(obj.attributes)
+                    x_states.push({ "id" : obj.id, "name" : obj.attributes.name})
                 }
+                console.log(x_states);
```

Comment:
@MatthewKennedy `console.log` :)

Reply:
That’s not the worst of it, when you put an address in and click continue it flashes up billing and shipping address requires country.

I’m assuming that the main ruby code is still expecting an integer for the country_id, and possibly other parts of the  code are expecting an integer country_id such as shipping method and coupon code. 

I’ve had a go but can’t seem to fix it, any pointers?

There’s still a load of tidying up to do for the code climate too.